tests/ui/const-generics/generic_const_exprs/const_kind_expr/issue_114151.rs

```diff
@@ -20,6 +20,7 @@ where
     //~^^^ ERROR: function takes 1 generic argument but 2 generic arguments were supplied
     //~^^^^ ERROR: unconstrained generic constant
     //~^^^^^ ERROR: unconstrained generic constant `{const expr}`
+    //~^^^^^^ ERROR: unconstrained generic constant `{const expr}`
```

Comment:
The test output is changed, but I think that this would be correct because `L + 1 + L` is actually nested const expr.

The difference might be came from the difference in some recursive visiting behaviour between `TypeWalker` and `Const::super_visit_with`;
https://github.com/rust-lang/rust/blob/52f8aec14c616387c5f793687f2d9026de6c78ca/compiler/rustc_middle/src/ty/walk.rs#L205-L236
https://github.com/rust-lang/rust/blob/52f8aec14c616387c5f793687f2d9026de6c78ca/compiler/rustc_middle/src/ty/structural_impls.rs#L762-L779
the former doesn't track down some inner values.
It might possible to getting the same behaviour by changing implementation of `impl TypeVisitor for WfPredicates`, but it would be unintuitive and since this output seems better - at least I think so 😅 - I think that it would be good in this way.